src/components/Card.js

```diff
@@ -1,11 +1,15 @@
 class Card {
-  constructor(cardData, cardSelector, handlePreviewClick) {
+  constructor(cardData, cardSelector, handlePreviewClick, handleDeleteClick, userId, handleUserLikes) {
     this._name = cardData.name;
     this._link = cardData.link;
-    this._likeButton = this;
+    this._cardId = cardData._id;
     this._deleteButton = this;
     this._cardSelector = cardSelector;
     this._handlePreviewClick = handlePreviewClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._likes = cardData.likes || [];
```

Comment:
`    this._likes = cardData.likes || [];
`

`cardData.likes` doesn't exist


You should define `this._isLiked`


```js

    this._isLiked = cardData.isLiked

```